packages/core/upload/server/services/extensions/content-manager/entity-manager.js

```diff
@@ -77,46 +90,28 @@ const addSignedFileUrlsToAdmin = () => {
   // We need to do this for create/update/delete/publish/unpublish too no?
   strapi.container
     .get('services')
-    .extend(`plugin::content-manager.entity-manager`, (entityManager) => {
-      const update = async (entity, body, uid) => {
-        const updatedEntity = await entityManager.update(entity, body, uid);
-        await signEntityMedia(updatedEntity, uid);
-        return updatedEntity;
-      };
-
-      const publish = async (entity, body, uid) => {
-        const publishedEntity = await entityManager.publish(entity, body, uid);
-        await signEntityMedia(publishedEntity, uid);
-        return publishedEntity;
-      };
-
-      const unpublish = async (entity, body, uid) => {
-        const unpublishedEntity = await entityManager.unpublish(entity, body, uid);
-        await signEntityMedia(unpublishedEntity, uid);
-        return unpublishedEntity;
-      };
-
-      const findOneWithCreatorRolesAndCount = async (id, uid) => {
-        // TODO: What if the entity is not found?
-        const entity = await entityManager.findOneWithCreatorRolesAndCount(id, uid);
-        await signEntityMedia(entity, uid);
-        return entity;
-      };
-
-      const findWithRelationCountsPage = async (opts, uid) => {
-        const entities = await entityManager.findWithRelationCountsPage(opts, uid);
-        await mapAsync(entities.results, async (entity) => signEntityMedia(entity, uid));
-        return entities;
-      };
-
-      return {
-        ...entityManager,
-        findOneWithCreatorRolesAndCount,
-        findWithRelationCountsPage,
-        update,
-        publish,
-        unpublish,
-      };
+    .extend('plugin::content-manager.entity-manager', (entityManager) => {
+      const functionsToExtend = [
+        'update',
+        'publish',
+        'unpublish',
+        'findOneWithCreatorRolesAndCount',
+        'findWithRelationCountsPage',
+      ];
+
+      const extendedFunctions = {};
+      functionsToExtend.reduce((acc, functionName) => {
+        acc[functionName] = async (...args) => {
+          const result = await entityManager[functionName](...args);
+          await fileSigningExtension(result, args.at(-1));
+
+          return result;
+        };
+
+        return acc;
+      }, extendedFunctions);
```

Comment:
My feeling is this bit of code is shorter but not clearer than before 🤔 What do you think?

Do you think we could find a way to put this inside the content-manager plugin? A method like (mapEntity) that can be extended from any plugin.

Maybe it's complicating ourselves for what we need.